src/utils.jl

```diff
@@ -114,3 +114,70 @@ function optimal_shrinkage!(X::AbstractArray{T, 2}) where T <: Number
     X .= U*Diagonal(S)*V'
     return
 end
+
+function savitzky_golay(x::AbstractVector{T}, windowSize::Integer, polyOrder::Integer; deriv::Integer=0, dt::Real=1.0) where T <: Number
```

Comment:
Can we use this with an additional argument to return the cropped data like in the example? 
I think this would be a nicer api for users :) 

Something like 
```julia
function savitzky_golay(x::AbstractVector{T}, windowSize::Integer, polyOrder::Integer; deriv::Integer=0, dt::Real=1.0, crop::Bool = true) where T <: Number

# Do stuff ...

if !crop
# Return the valid midsection
	return y[2*halfWindow+1:end-2*halfWindow]
else
	return x[2*halfWindow+1:end-2*halfWindow],  y[2*halfWindow+1:end-2*halfWindow]
end
```

Reply:
Thanks! Good suggestion! I'll implement it later today.

I was planning to add the feature of replacing the portion that we are cropping for its interpolation (or its derivative), as it is the default option in matlab's and scipy's savitzky golay functions. That way the quality in the borders improves and we wouldn't necessarily have to crop the resulting arrays. But until I implement that (and even with the interpolation feature) 
 I think it's a good idea to have the option for cropping inside the function.

Reply:
:+1:  The replacing part makes sense and would be awesome to include.